index.rst

```diff
@@ -5,77 +5,331 @@
+The following are the key functionality we are looking for in the ``Catcher``.
 
-.. _Flux: https://docs.influxdata.com/flux/v0.x/
-.. _InfluxDB task: https://docs.influxdata.com/influxdb/v2.2/process-data/get-started/
+Commandable SAL Component
+-------------------------
 
+The ``Catcher`` needs to interact with the Rubin Observatory Control System, both by reading telemetry from the system and by commanding other CSCs (see :ref:`introduction-process-image-data`).
+We also expect to be able to interact with the ``Catcher`` in a similar way we interact with other components on the system, such that we can start, configure, stop and redeploy it the same way we do with other components.
 
-`Flux`_ is a new functional data scripting language that is designed to process data streams.
-It contains extensive libraries that allows users to express complex operations, compatible with what can be done with programming languanges.
-Furthermore, `InfluxDB task`_ provides a way to schedule `Flux`_ scripts.
+Furthermore, being a CSC we also expect it to be written using Python/salobj, which is a broadly used framework in the project. 
+This ensures we have enough knowledge in the project to maintain and support the ``Catcher`` development.
 
+One thing we must consider in this case is whether the ``Catcher`` CSC should be a single CSC that handles all conditions, or an indexed CSC were each instance handles a single case/rule.
+Using our experience with the ``Watcher`` it seems like having an indexed CSC would be preferable.
+Some reasons for this are:
 
-Each Catcher "rule" or "alarm" would be comprised of an `InfluxDB task`_ to execute a user defined `Flux`_ script periodically.
-The `Flux`_ script performs a set of actions/calculations based on EFD data, additional processes/tasks can be triggered depending on the results of these calculations and results can be stored in the EFD for future reference.
+*  A single CSC means there is a single event for each condition.
+   If the CSC is handling multiple conditions the information ends up getting "lost" in the data queue and it is not easily digestible by LOVE and other systems.
+   Whereas, if each instance of the ``Catcher`` handles a single condition/rule, events are well separated for each condition.
 
-Below we describe how some of the Catcher functionality would be implemented.
+*  In the single CSC scenario, adding/removing/updating a particular rule requires restarting the entire system.
+   In the indexed component scenario, it only requires bringing in a new instance or removing/updating an existing one.
 
-Alarms
-------
+To control the configuration of each ``Catcher`` instance, we can adopt an indexed-based configuration format, used by some CSCs in the system already.
```

Comment:
you have an example?
We've done a pretty stupid system (one file per index) with the electrometer if I remember correctly...